### ISSUE TITLE
Add option to disable PCH

### ISSUE THREAD

**Issue by marxin** [telegramdesktop/tdesktop#7054]
### Is your feature request related to a problem?
GCC has a problem with pre-compiler headers:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93387

### Describe the solution you'd like
Would it be possible to disable PCH during build?

**Comment by john-preston** [comment]
@marxin For now c++2a is disabled for gcc.

**Comment by marxin** [comment]
> @marxin For now c++2a is disabled for gcc.

You mean in the official build?

**Comment by john-preston** [comment]
@marxin In the CMake files, see https://github.com/desktop-app/cmake_helpers/commit/67cf2a5abdb01658c1cf852b29e25808dcc02c56

**Comment by marxin** [comment]
Ah, I see. Have you created a GCC issue for that? Do you have a test-case where you see a problematic `c++2a` implementation in GCC 9?

**Comment by john-preston** [comment]
@marxin Well, you created it, this is the problematic implementation — it crashes the compiler in range-v3 library in some cases.

**Comment by marxin** [comment]
Sure. Then, I would be happy to test it with current GCC master, but I see:
```
[   88s] /usr/bin/c++  -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DDESKTOP_APP_USE_PACKAGED_FONTS -DNDEBUG -DTGVOIP_USE_DESKTOP_DSP -DTGVOIP_USE_INSTALLED_OPUS -DWEBRTC_APM_DEBUG_DUMP=0 -DWEBRTC_LINUX -DWEBRTC_NS_FLOAT -DWEBRTC_POSIX -I../Telegram/ThirdParty/libtgvoip -I../Telegram/ThirdParty/libtgvoip/webrtc_dsp -isystem /usr/include/opus -O3 -DNDEBUG   -Wno-unknown-pragmas -Wno-error=sequence-point -Wno-error=unused-result -fno-strict-aliasing -pipe -Wall -W -fPIC -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function -Wno-switch -Wno-comment -Wno-unused-but-set-variable -Wno-missing-field-initializers -Wno-sign-compare -Wno-attributes -Wno-parentheses -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=class-memaccess -std=gnu++2a -MD -MT Telegram/CMakeFiles/lib_tgvoip.dir/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc.o -MF Telegram/CMakeFiles/lib_tgvoip.dir/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc.o.d -o Telegram/CMakeFiles/lib_tgvoip.dir/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc.o -c ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc
[   88s] In file included from ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:16:
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h: In instantiation of 'class absl::InlinedVector<short int*, 8, std::allocator<short int*> >':
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:123:30:   required from 'int webrtc::PushResampler<T>::Resample(const T*, size_t, T*, size_t) [with T = short int; size_t = long unsigned int]'
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:148:16:   required from here
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:93:9: error: no type named 'pointer' in 'using allocator_type = class std::allocator<short int*>' {aka 'class std::allocator<short int*>'}
[   88s]    93 |   using pointer = typename allocator_type::pointer;
[   88s]       |         ^~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:94:9: error: no type named 'const_pointer' in 'using allocator_type = class std::allocator<short int*>' {aka 'class std::allocator<short int*>'}
[   88s]    94 |   using const_pointer = typename allocator_type::const_pointer;
[   88s]       |         ^~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:95:9: error: no type named 'reference' in 'using allocator_type = class std::allocator<short int*>' {aka 'class std::allocator<short int*>'}
[   88s]    95 |   using reference = typename allocator_type::reference;
[   88s]       |         ^~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:96:9: error: no type named 'const_reference' in 'using allocator_type = class std::allocator<short int*>' {aka 'class std::allocator<short int*>'}
[   88s]    96 |   using const_reference = typename allocator_type::const_reference;
[   88s]       |         ^~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:101:9: error: no type named 'pointer' in 'class std::allocator<short int*>'
[   88s]   101 |   using reverse_iterator = std::reverse_iterator<iterator>;
[   88s]       |         ^~~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:102:9: error: no type named 'const_pointer' in 'class std::allocator<short int*>'
[   88s]   102 |   using const_reverse_iterator = std::reverse_iterator<const_iterator>;
[   88s]       |         ^~~~~~~~~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc: In instantiation of 'int webrtc::PushResampler<T>::Resample(const T*, size_t, T*, size_t) [with T = short int; size_t = long unsigned int]':
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:148:16:   required from here
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:128:69: error: 'class absl::InlinedVector<short int*, 8, std::allocator<short int*> >' has no member named 'data'
[   88s]   128 |   Deinterleave(src, src_length_mono, num_channels_, source_pointers.data());
[   88s]       |                                                     ~~~~~~~~~~~~~~~~^~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:143:35: error: 'class absl::InlinedVector<short int*, 8, std::allocator<short int*> >' has no member named 'data'
[   88s]   143 |   Interleave(destination_pointers.data(), dst_length_mono, num_channels_, dst);
[   88s]       |              ~~~~~~~~~~~~~~~~~~~~~^~~~
[   88s] In file included from ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:16:
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h: In instantiation of 'class absl::InlinedVector<float*, 8, std::allocator<float*> >':
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:123:30:   required from 'int webrtc::PushResampler<T>::Resample(const T*, size_t, T*, size_t) [with T = float; size_t = long unsigned int]'
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:149:16:   required from here
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:93:9: error: no type named 'pointer' in 'using allocator_type = class std::allocator<float*>' {aka 'class std::allocator<float*>'}
[   88s]    93 |   using pointer = typename allocator_type::pointer;
[   88s]       |         ^~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:94:9: error: no type named 'const_pointer' in 'using allocator_type = class std::allocator<float*>' {aka 'class std::allocator<float*>'}
[   88s]    94 |   using const_pointer = typename allocator_type::const_pointer;
[   88s]       |         ^~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:95:9: error: no type named 'reference' in 'using allocator_type = class std::allocator<float*>' {aka 'class std::allocator<float*>'}
[   88s]    95 |   using reference = typename allocator_type::reference;
[   88s]       |         ^~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:96:9: error: no type named 'const_reference' in 'using allocator_type = class std::allocator<float*>' {aka 'class std::allocator<float*>'}
[   88s]    96 |   using const_reference = typename allocator_type::const_reference;
[   88s]       |         ^~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:101:9: error: no type named 'pointer' in 'class std::allocator<float*>'
[   88s]   101 |   using reverse_iterator = std::reverse_iterator<iterator>;
[   88s]       |         ^~~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/absl/container/inlined_vector.h:102:9: error: no type named 'const_pointer' in 'class std::allocator<float*>'
[   88s]   102 |   using const_reverse_iterator = std::reverse_iterator<const_iterator>;
[   88s]       |         ^~~~~~~~~~~~~~~~~~~~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc: In instantiation of 'int webrtc::PushResampler<T>::Resample(const T*, size_t, T*, size_t) [with T = float; size_t = long unsigned int]':
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:149:16:   required from here
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:128:69: error: 'class absl::InlinedVector<float*, 8, std::allocator<float*> >' has no member named 'data'
[   88s]   128 |   Deinterleave(src, src_length_mono, num_channels_, source_pointers.data());
[   88s]       |                                                     ~~~~~~~~~~~~~~~~^~~~
[   88s] ../Telegram/ThirdParty/libtgvoip/webrtc_dsp/common_audio/resampler/push_resampler.cc:143:35: error: 'class absl::InlinedVector<float*, 8, std::allocator<float*> >' has no member named 'data'
[   88s]   143 |   Interleave(destination_pointers.data(), dst_length_mono, num_channels_, dst);
[   88s]       |              ~~~~~~~~~~~~~~~~~~~~~^~~~
```

**Comment by Aokromes** [comment]
fixed on 2.1
